### PR TITLE
34: Created settings file to store magic variables (company collectio…

### PR DESCRIPTION
**Pull request by c0nsy**
Created GET by id endpoint, implemented functionality in core

added a settings file and a way for the team to get the settings in an easy way, importing settings and calling get_settings()

we do this because we will not hardcode magic values in the code base, if they change we just need to update one spot: the settings file

## Describe your changes
Endpoint that returns a company record depending on the document id (str).

## Issue ticket number and link
[34](https://github.com/Hermes-Marketing/hermes/issues/34)
## How to test changes
switch to branch, navigate to fastapi docs, trigger get by id endpoint with a valid id (str). Some valid ones, 0,10,100,2,3...

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [] If it is a core feature, I have added thorough tests.
- [] Do we need to implement analytics?

### CODE REVIEW

File: backend/app/api/v1/endpoints/company.py [hunk 1]
```diff
@@ -26,3 +26,16 @@ async def get_all_companies(db_session=Depends(get_db)):
     """
     
     return CompanyRepository(db_session).get_all_companies()
+
+@router.get("/{id}", status_code=status.HTTP_200_OK, response_model=Company)
+async def get_company_by_id(id: str, db_session=Depends(get_db)):
```
Comment (xbnrxout) [inline backend/app/api/v1/endpoints/company.py:31]: heh depends. Maybe ill get the records, maybe i wont. depends

File: backend/app/api/v1/endpoints/company.py [hunk 2]
```diff
@@ -26,3 +26,16 @@ async def get_all_companies(db_session=Depends(get_db)):
...
+    """
```
Comment (xbnrxout) [inline backend/app/api/v1/endpoints/company.py:32]: While block comments are good, and are a flavour of documentation. There is something called the clean code principle. If you name your variables and functions correctly. Code speaks for itself. IF its a complex algo, large return block from call, or some data dump response, then you do block comments.

File: backend/app/config/settings.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# All Constants defined here
+
+from pydantic import BaseSettings
+
+
+class Settings(BaseSettings):
+    # Collections
+    COMPANY_COLLECTION: str = "companies"
```
Comment (xbnrxout) [inline backend/app/config/settings.py:8]: What is this for?

File: backend/app/core/company.py [hunk 4]
```diff
@@ -7,10 +7,53 @@
...
+
+        Args:
+            id (str): The document id of the company to retrieve
+
+        Returns:
+            Company: The company object from the collection companies
+
+        Raises:
+            Exception: If the company does not exist
+        """
```
Comment (xbnrxout) [inline backend/app/core/company.py:28]: Same as previous comment about block comments.

File: backend/app/core/company.py [hunk 5]
```diff
@@ -7,10 +7,53 @@
 from typing import List
 from app.core.main import AppRepository
 from app.models.company import Company
+from app.config.settings import get_settings
+from fastapi import HTTPException, status
 import logging
-
+settings = get_settings()
```
Comment (xbnrxout) [inline backend/app/core/company.py:13]: nice naming 💯

File: backend/app/core/company.py [hunk 6]
```diff
@@ -7,10 +7,53 @@
...
+
+        Returns:
+            Company: The company object from the collection companies
+
+        Raises:
+            Exception: If the company does not exist
+        """
+        doc = self.db.collection(settings.COMPANY_COLLECTION).document(id).get()
+        if not doc.exists:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Company not found")
```
Comment (xbnrxout) [inline backend/app/core/company.py:31]: nice. THis is the correct status. 💯

File: backend/app/core/company.py [hunk 7]
```diff
@@ -23,7 +66,7 @@ def get_all_companies(self) -> List[Company]:
         """
         companies = []
         try:
-            docs = self.db.collection('companies').stream()
+            docs = self.db.collection(settings.COMPANY_COLLECTION).stream()
```
Comment (xbnrxout) [inline backend/app/core/company.py:69]: im not sure of the point

**Review by xbnrxout — APPROVED** [review]

File: backend/app/config/settings.py (hunk 3 above)
Comment (c0nsy) [reply backend/app/config/settings.py:8]: as we grow in complexity, we will organize and store all magic variables here that pertain to the backend

File: backend/app/api/v1/endpoints/company.py (hunk 1 above)
Comment (c0nsy) [reply backend/app/api/v1/endpoints/company.py:31]: lol, depends if you have the db session!

File: backend/app/core/company.py (hunk 7 above)
Comment (c0nsy) [reply backend/app/core/company.py:69]: magic variable

**Pull request merged**